### PR TITLE
[bitnami/metallb] kube version support

### PR DESCRIPTION
**Pull request by my-git9**
### Description of the change
Set a minimum kubeVersion in Chart.yaml

When I install metallb by chart, the output:

```
$ helm install metallb -nkube-system ./ --dry-run
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(CustomResourceDefinition.spec.versions[0]): unknown field "deprecated" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion, ValidationError(CustomResourceDefinition.spec.versions[0]): unknown field "deprecationWarning" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion, ValidationError(CustomResourceDefinition.spec.versions[1]): unknown field "deprecated" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion, ValidationError(CustomResourceDefinition.spec.versions[1]): unknown field "deprecationWarning" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion]
```

Find the  field of  "deprecated"  has been supported from 1.19 version for k8s
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-deprecation

### Applicable issues

  - fixes #

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/metallb/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,9 @@ sources:
   - https://github.com/metallb/metallb
   - https://github.com/bitnami/containers/tree/main/bitnami/metallb
   - https://metallb.universe.tf
-version: 4.1.10
+version: 4.1.11
+
+# The optional kubeVersion field can define semver constraints on supported Kubernetes versions.
+# Helm will validate the version constraints when installing the chart and fail if the cluster
+# runs an unsupported Kubernetes version.
+kubeVersion: ">= 1.19.0-0"
```
Comment (recena) [inline bitnami/metallb/Chart.yaml:36]: @javsalgar Should we consider it for all charts?

Comment (carrodher) [reply bitnami/metallb/Chart.yaml:36]: We are already adding it to those Helm charts where there is a specific k8s resource that was deprecated in a specific Kubernetes version:
```console
$ ag 'kubeVersion' */Chart.yaml
contour-operator/Chart.yaml
20:kubeVersion: '>= 1.21.0-0'

kubeapps/Chart.yaml
27:kubeVersion: ">=1.21.0-0"

kube-prometheus/Chart.yaml
28:kubeVersion: '>= 1.16.0-0'

nginx-ingress-controller/Chart.yaml
22:kubeVersion: '>= 1.20.0-0'

rabbitmq-cluster-operator/Chart.yaml
21:kubeVersion: '>= 1.19.0-0'

sealed-secrets/Chart.yaml
18:kubeVersion: '>=1.16.0-0'
```

Comment (carrodher) [inline bitnami/metallb/Chart.yaml:36]: Could you please place the new parameter in alphabetical order (between `keywords` and `maintainers`)?
In the same way, we are not adding that default comment in other cases, see some examples at
- https://github.com/bitnami/charts/blob/main/bitnami/kube-prometheus/Chart.yaml#L28
- https://github.com/bitnami/charts/blob/main/bitnami/rabbitmq-cluster-operator/Chart.yaml#L21

Comment (my-git9) [reply bitnami/metallb/Chart.yaml]: Updated

**Review by migruiz4 — APPROVED** [review]
Hi @my-git9,

Thank you very much for your contribution! Looks good to me!

**Pull request merged**